pandas/tests/frame/test_query_eval.py

```diff
@@ -1038,3 +1038,34 @@ def test_invalid_type_for_operator_raises(self, parser, engine, op):
 
         with pytest.raises(TypeError, match=msg):
             df.eval('a {0} b'.format(op), engine=engine, parser=parser)
+
+
+class TestDataFrameQueryBacktickQuoting(object):
+
+    def setup_method(self, method):
+        self.df = DataFrame({'A': [1, 2, 3],
+                             'B B': [3, 2, 1],
+                             'C C': [4, 5, 6]})
+
+    def teardown_method(self, method):
+        del self.df
+
+    def test_single_backtick_variable_query(self):
+        res = self.df.query('1 < `B B`')
```

Comment:
you would need to test on column names that have _ already in them; your patch will likely fail (also mixed, IOW one has spaces, one has the same name when composed with _)

Reply:
Included more tests.